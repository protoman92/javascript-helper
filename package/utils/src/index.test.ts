import { deepClone, stripLeadingSlash } from ".";

describe("General utilities", () => {
  it("Strip leading slashes should work correctly", async () => {
    // Setup
    // When
    // Then
    expect(stripLeadingSlash("////abc")).toEqual("abc");
  });

  it("Deep clone should work correctly", () => {
    // Setup
    // When
    const deepCloned = deepClone({ a: { b: { c: { d: 1, e: "1" } } } })(
      (...[, value]) => {
        if (typeof value === "number") return value.toString();
        return value;
      },
      (...[, value]) => {
        if (typeof value === "string") return parseInt(value);
        return value;
      }
    )(
      (...[, value]) => {
        if (typeof value === "number") return value * 10;
        return value;
      },
      (...[, value]) => {
        if (typeof value === "number") return value.toString();
        return value;
      }
    );

    // Then
    expect(deepCloned).toMatchSnapshot();
  });
});
