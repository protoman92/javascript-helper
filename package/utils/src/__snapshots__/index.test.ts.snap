// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General utilities Deep clone should work correctly 1`] = `
{
  "a": {
    "b": {
      "c": {
        "d": "10",
        "e": "10",
      },
    },
  },
}
`;
